src/test/java/gil/server/StaticFileOptionsControllerTest.java

```diff
@@ -0,0 +1,38 @@
+package gil.server;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+public class StaticFileOptionsControllerTest {
+    StaticFileOptionsController staticFileOptionsController = new StaticFileOptionsController();
+
+    @Test
+    public void shouldRespondToAnOPTIONSRequestForAnExistingStaticFile() {
+        Request request = new Request();
+        Response response = new Response();
+        request.setMethod("OPTIONS");
+        request.setHttpVersion("HTTP/1.1");
+        request.setURI("/file%20with%20space.txt");
+        String expectedHeaderFields = "Allow: OPTIONS, GET";
+
+        staticFileOptionsController.options.apply(request, response);
+
+        assertEquals("200", response.getStatusCode());
+        assertEquals("OK", response.getReasonPhrase());
+        assertEquals(expectedHeaderFields, response.getAllow());
+    }
+
+    @Test
+    public void shouldRespondWith404ToAnOPTIONSRequestForANonExistingStaticFile() {
+        Request request = new Request();
+        Response response = new Response();
+        request.setHttpVersion("HTTP/1.1");
+        request.setMethod("OPTIONS");
+        request.setURI("/mystery-file.txt");
```

Comment:
For request and responses, how come you chose to use setters instead of using a constructor that takes in multiple parameters?

Reply:
My use of setters was inspired by the builder pattern. They are very useful for the Controllers because I wanted them to accept a Response as an argument and if I replaced them with a new object inside of the controller function the reference to the object inside the controller body would be to a different object than the one being passed as an argument. Does this make sense? What do you think?